lib/modules/manager/vendir/artifacts.ts

```diff
@@ -0,0 +1,123 @@
+import is from '@sindresorhus/is';
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import {
+  getSiblingFileName,
+  readLocalFile,
+  writeLocalFile,
+} from '../../../util/fs';
+import { getRepoStatus } from '../../../util/git';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  logger.debug(`vendir.updateArtifacts(${packageFileName})`);
+
+  const isLockFileMaintenance = config.updateType === 'lockFileMaintenance';
+  const isUpdateOptionAddVendirArchives = config.postUpdateOptions?.includes(
+    'vendirUpdateSubChartArchives',
+  );
+
+  if (
+    !isLockFileMaintenance &&
+    (updatedDeps === undefined || updatedDeps.length < 1)
+  ) {
+    logger.debug('No updated vendir deps - returning null');
+    return null;
+  }
+
+  const lockFileName = getSiblingFileName(packageFileName, 'vendir.lock.yml');
+  const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
```

Comment:
I think this will throw for non existing files, so check lockfile name before

Reply:
> I think this will throw for non existing files, so check lockfile name before

Check if the lockFile exists? Not sure what this comment means.

Reply:
Added a check for the lockFile existing